magit-filenotify.el

```diff
@@ -56,9 +56,46 @@
              collect (expand-file-name dir))
     :test #'string=)))
 
-(defvar magit-filenotify-data (make-hash-table)
+;; Use #'equal as test because watch-descriptors aren't always integers.  With
+;; inotify, descriptors can be lists like (17).
+(defvar magit-filenotify-data (make-hash-table :test #'equal)
   "A hash table to map watch-descriptors to a list (DIRECTORY STATUS-BUFFER).")
 
+(defvar magit-filenotify--idle-timer nil
+  "Timer which will refresh buffers when emacs becomes idle.")
+
+(defvar magit-filenotify-idle-delay 1.57
```

Comment:
Rüdiger Sonderfeld <notifications@github.com> writes:

>> +(defvar magit-filenotify--idle-timer nil
>> +  "Timer which will refresh buffers when emacs becomes idle.")
>> +
>> +(defvar magit-filenotify-idle-delay 1.57
>
> Shouldn't this be a defcustom?

Yes, that wouldn't hurt.  I'll make it so.

> Why 1.57s?

Seemed like a good compromise.  It should be longer than a typical
compilation of a file takes.  For example, if it was 1 but the
compilation of some file takes 1.1 seconds on average (and we assume
sequential compilation of many files in the repo), then basically we'd
again have one refresh per file update.

Well, and it is generally advisable to use completely odd timer values
(therefore 1.57 and not 1.5) because that makes it more unlikely that
many timers are run at the same time.

Bye,
Tassilo